src/snowflake/snowpark/modin/plugin/_internal/frame.py

```diff
@@ -415,7 +415,9 @@ def index_columns_index(self) -> pd.Index:
             index_type = TypeMapper.to_pandas(
                 self.quoted_identifier_to_snowflake_type()[index_identifier]
             )
-            ret = pd.Index(
```

Comment:
we can return native_pd.Index in this method so you don't need to make those changes. The ideas is to keep `df.columns` eager.

Reply:
`df.columns` does not call this, it calls `data_columns_index`

Reply:
oh my bad!